### DISCUSSION TITLE
Does digital have 4-bit full adder?

### DISCUSSION THREAD

**Discussion by 4book** [hneemann/Digital#1217] [Q&A] [answered]
I found that we can select the data bits of an adder, but after changing the data bits, the places to input are still 3. So how can I use a wire to input a 4-bit data?

**Comment by hneemann** [comment, marked as answer]
To keep more complex circuits clear, buses are supported: One wire can represent several bits. 
This is used in the adder: If four bits are selected, a 4 bit bus is connected to the inputs and the sum output. 
See also #378

**Reply by 4book** [reply]
Thx！

**Comment by mengstr** [comment]
As an alternative if you absolutely want to have a 4-bit Adder with all the connections you could use the 7483 or 74283 logic ICs that are available in Digital and include it in your design.

If you right click it you can then choose between it looking like a DIP-IC or just a regular rectangle.

<img width="667" alt="Screenshot 2023-10-12 at 20 26 52" src="https://github.com/hneemann/Digital/assets/325326/d5609a79-a5b6-4b19-9692-746309cf392a">
